mantidproject/mantid
# Fix workspace property handling in MaskDetectors

The algorithm should not call `setProperty` on an *InOut* workspace property since it can invalidate a variable in python.

**Description of work.**

**Report to:** [nobody].

**To test:**

Check that the script below runs without reporting any errors. The script is stolen from the issue description. The test data is available in the unit test data directories, so make sure that these are included in Mantid's data search directories.
```python
mtd.clear()
ws1 = CreateSampleWorkspace()
maskWS1, detectorList = ExtractMask(ws1)
MaskDetectors(maskWS1, DetectorList='1', ComponentList='bank1')
print(maskWS1)

ws2 = DirectILLCollectData('ILL/IN5/104007.nxs')
maskWS2, detList = ExtractMask(ws2)
MaskDetectors(maskWS2, DetectorList='1', ComponentList='tube_1')
print(maskWS2)
```

Fixes #23600.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
